drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/fetch_clients_with_active_referrals_job.rb

```diff
@@ -27,14 +27,16 @@ def perform
         user: system_user,
       )
 
-      Hmis::Hud::Client.
-        where(id: client_ids).
-        each do |client|
-          client.custom_data_elements.
-            where(data_element_definition: cded).
-            first_or_create!(value_boolean: true, data_source: data_source, user: system_user).
-            update!(value_boolean: true)
-        end
+      Hmis::Hud::CustomDataElement.import(
+        Hmis::Hud::Client.
+          where(id: client_ids).
+          map do |client|
+            cde = client.custom_data_elements.where(data_element_definition: cded).first_or_create!(data_source: data_source, user: system_user, value_boolean: true)
+            cde.value_boolean = true
+            cde
+          end,
+        on_duplicate_key_update: { conflict_target: [:id], columns: [:value_boolean] },
+      )
```

Comment:
Isn't this still an N+1 query @ZachCMP ?
```
        Hmis::Hud::Client.
          where(id: client_ids).
          map do |client| # 1 query to get all clients, convert ActiveRecord scope to an array
            # 1 query (per array item) to lookup custom data elements
            cde = client.custom_data_elements.where(data_element_definition: cded)....
```


I'm not actually looking at the db queries that are made, so let me know if I'm missing something.


I was thinking something more like this (untested). There is no need to make 1 query per client if we just do an import.
```
    attributes = client_ids.map do |client_id|
      {
        owner_type: 'Hmis::Hud::Client',
        owner_id: client_id,
        data_element_definition_id: cded.id,
        value_boolean: true,
        data_source_id: data_source_id,
        user_id: user_id,
      }
    end
    Hmis::Hud::CustomDataElement.import(attributes, on_duplicate_key_update: { conflict_target: [:owner_id, :owner_type, :data_element_definition_id], columns: [:value_boolean, :user_id] })
```